src/Files.App/Utils/Storage/Operations/FileOperationsHelpers.cs

```diff
@@ -1018,5 +1018,26 @@ protected override void Dispose(bool disposing)
 				}
 			}
 		}
+
+		public static string GetIncrementalName(bool overWriteOnCopy, string? filePathToCheck, string? filePathToCopy)
+		{
+			if (filePathToCheck == null)
+				return null;			
+
+			if ((!Path.Exists(filePathToCheck)) || overWriteOnCopy || filePathToCheck == filePathToCopy)
+				return Path.GetFileName(filePathToCheck);
+
+			int index = 2;
+			string filePath = filePathToCheck;
+			if (Path.HasExtension(filePathToCheck))
+				filePath = filePathToCheck.Substring(0, filePathToCheck.LastIndexOf("."));
+
+			Func<int, string> genFilePath = x => string.Concat([filePath, " (", x.ToString(), ")", Path.GetExtension(filePathToCheck)]);
+
+			while (Path.HasExtension(filePathToCheck) ? File.Exists(genFilePath(index)) : Path.Exists(genFilePath(index)))			
```

Comment:
```suggestion
			while (Path.Exists(genFilePath(index)))			
```
Some files don't have extensions, and it is enough to just use `Path.Exists()` in the first place.